### ISSUE TITLE
Formatting style for editor.formatOnSave is different.

### ISSUE THREAD

**Issue by timblakely** [xaverh/vscode-clang-format-provider#24]
In 1.6.0, my settings.json contains {"clang-format.style": "google"}. If I right-click inside of a file and choose format, the formatting is correctly applied in the "google" style. If I enable editor.formatOnSave, the formatting style is changed when formatting is applied during saving. 

Right-click->Format Code formatting:

```
class Foo {
  constructor() {
    let foo = {bar: 1};
    Promise.resolve(1)
        .then(result => {
          console.log(result);
        })
        .then(() => 2)
        .then(v => {
          console.log('received ', v);
        });
  }
}
```

Saving (applying editor.formatOnSave) formatting:

```
class Foo {
  constructor() {
    let foo = { bar: 1 };
    Promise.resolve(1)
      .then(result => {
        console.log(result);
      })
      .then(() => 2)
      .then(v => {
        console.log('received ', v);
      });
  }
}
```

Note the change in spacing in the "let foo" line and the tab formatting of the Promise's .then() functions.

**Comment by timblakely** [comment]
I believe this may be related to https://github.com/Microsoft/vscode/issues/11609

**Comment by xaverh** [comment]
Is the formatting that you get, when you have editor.formatOnSave enabled the default formatter's? That is, do you get that very formatting if you disable the extension entirely?

**Comment by timblakely** [comment]
Yup, that's correct. Uninstalling the clang-format extension still results in the latter formatting.

**Comment by timblakely** [comment]
Ah, it appears the autosave feature calls provideDocumentRangeFormattingEdit, whereas the editor.action.format uses provideDocumentFormattingEdit. For some reason, VSCode opts to use clang-format for the latter, but chooses to use the built-in typescript formatter in /usr/share/code/resources/app/extensions/typescript/out/features/formattingProvider.js for the former.

Not sure the best way around that, other than waiting for Microsoft/vscode#11609 to settle...

**Comment by xaverh** [comment]
This should be solved with Visual Studio Code 1.7 since you can now simply deactivate the built-in formatters.

**Comment by timblakely** [comment]
Yup, can confirm using 1.7 doesn't show this behavior anymore. Thanks for looking into it.
